shared/test/showcase/echo/echo_test.rb

```diff
@@ -18,23 +18,19 @@
 require "google/showcase/v1beta1/echo"
 require "grpc"
 
-class EchoTest < ShowcaseTest
-  def setup client=nil
-    @client = client
+class EchoGRPCTest < ShowcaseTest
+  def setup
+    @client = new_echo_client
   end
 
-  def test_echo
-    return unless @client
+  def test_universe_domain
+    assert_equal "googleapis.com", @client.universe_domain
```

Comment:
I would like if there was a test for universe_domain substitution -- configure echo with a different universe_domain and see the endpoint change. Can also be done in generated tests

Reply:
Can't add those tests to showcase because its endpoint is locked to localhost (it's not a real service). I added a set of tests for vision_v1 to test universe_domain and endpoint logic from configuration and environment variables.

Reply:
Great!